### PR TITLE
Global Query Params from URL

### PR DESCRIPTION
**Pull request by brumik**
TODO:
* [x] Redirect converts
* [x] export a function something like: `redirectWithQuery(Path.jobExplorer, attributes, namespace?)`
* [x] Update the tests
* [x] Bring back qs tests for the library and update them
  * Could be tested trough the `useRedirect` function
* [x] *BUG* Toolbar stopped to update the checkboxes on click, need investigation and fix
* [x] **WAITING FOR**: At least @jlmitch5 @h-kataria and @skochay330 went trough the UI to see if nothing is broken 

Followup:
* Convert the query param to TS

### CODE REVIEW

File: src/Components/Toolbar/Toolbar.tsx [hunk 1]
```diff
@@ -62,14 +54,11 @@ const FilterableToolbar: FunctionComponent<Props> = ({
...
     setQueryParams(key, value);
-    // Todo: this is temporary solution This call probably will not be here.
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
-    handleSearch(key, value, qsConfig, history);
   };
 
   return (
     <Toolbar
-      id={`${qsConfig.namespace}-filterable-toolbar-with-chip-groups`}
+      id={`${'namespace' ?? ''}-filterable-toolbar-with-chip-groups`}
```
Comment (brumik) [inline src/Components/Toolbar/Toolbar.tsx:61]: TODO

**Comment by h-kataria** [comment]
@brumik when i change the page size on the page to show 10 items on the screen and then press browser refresh button, it resets items to 5 items per page

File: src/Utilities/constants.js [hunk 2]
```diff
@@ -55,7 +55,8 @@ export const roi = {
     sort_order: 'desc',
     start_date: null,
     end_date: null,
-    limit: 25,
+    limit: '25',
+    ofset: '0',
```
Comment (h-kataria) [inline src/Utilities/constants.js:59]: typo

**Comment by h-kataria** [comment]
@brumik i am not sure if it is my environment, but the filters in the url does not seem to be working, when i apply filters and then press refresh in browser all filters get cleared from the screen. this was working earlier, maybe a recent change broke this. Can you verify?

**Comment by brumik** [comment]
@h-kataria I am sorry, it seems like I removed one important line when trying to figure out the browser back button. It is now back.

**Review by h-kataria — APPROVED** [review]

**Review by jlmitch5 — APPROVED** [review]
I have done quite a bit of clicking around in the UI.  I can't find any bugs or issues.  This is fantastic work and should fix a lot of things for us, great job @brumik

**Review by jlmitch5 — APPROVED** [review]

**Pull request merged**